lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java

```diff
@@ -164,7 +164,7 @@ public static ColumnReader create(PrimitiveField field, DateTimeZone timeZone, b
 
         return switch (primitiveType) {
             case BOOLEAN -> new BooleanColumnReader(field);
-            case INT32 -> createDecimalColumnReader(field).orElse(new IntColumnReader(field));
+            case INT32 -> createDecimalColumnReader(field).orElseGet(() -> new IntColumnReader(field));
```

Comment:
You might want to wait with those until @raunaqmorarka finished messing up with Parquet Reader

Reply:
This is a trivial change, but @raunaqmorarka let me know if this is a problem.

Reply:
reverting

Reply:
It's fine either way, not much going on in these constructors.